Add Hello World output feature, issue #1

haietza opened the pull request:
Added Greeting class to print "Hello, World!" to console, per feature request from customer on Issue #1. Also included makefile for build management on project going forward.

haietza committed:
Add Hello World output feature, issue #1

fenwickjb commented:
Would be better if testing were automated and not relying on humans who are error-prone. ;-)
Check this out...
http://stackoverflow.com/questions/1119385/junit-test-for-system-out-println

Also, it might help if you assign me as a "reviewer" (see panel to the right).

haietza committed:
Add automated testing

haietza commented:
Added automated testing for the Hello World feature. Thank you for the reference :).

fenwickjb commented on GreetingTest.java:
```diff
@@ -0,0 +1,28 @@
+
+    @After
+    public void tearDown() {
+        System.setOut(null);
```
Basically, this "breaks" System.out after the test. The test should configure everything it needs but then restore the world to the way it was before...

fenwickjb commented on GreetingTest.java:
```diff
@@ -0,0 +1,28 @@
+    @Test
+    public void testV1() {
+        Greeting greeting = new Greeting();
+        greeting.sayHello();
```
This is okay, but just FYI you can call the main method directly yourself here instead of using an instance method:
Greeting.main(null);
which assumes you are not using the argument...

fenwickjb commented:
I learned something doing this too! By going into the file and entering a review comment!

haietza committed:
Fix restoring System.out

haietza commented:
Fixed restoring System.out after test and modified class to use main method. Thank you for the pointers - I did not know that about calling main directly. The review comments specific to certain sections of code is certainly very helpful!

fenwickjb commented on GreetingTest.java:
```diff
@@ -16,13 +16,13 @@ public void setUp() {
     @After
     public void tearDown() {
-        System.setOut(null);
+        PrintStream originalOut = System.out;
```
But isn't this Ssytem.out the thing you changed it to in the setup? The originalOut should be a test class field set in setUp before the change and then restored here. Almost there!

haietza replied:
No, I changed the System.out to a PrintStream/ByteArrayStream in setUp; but I see that I should probably save the originalOut during the setUp before the test. Is the changed file what you were looking for? Thank you!

haietza committed:
Move setting of original System.out to setUp
